var/spack/repos/builtin/packages/leveldb/package.py

```diff
@@ -13,7 +13,9 @@ class Leveldb(MakefilePackage):
 
     homepage = "https://github.com/google/leveldb"
     url      = "https://github.com/google/leveldb/archive/v1.20.tar.gz"
+    git      = "https://github.com/google/leveldb.git"
 
+    version('master', branch='master')
```

Comment:
The convention we've been pushing for is `version('develop', branch='master')` since it represents the abstract concept of the current development version of a package and less the implementation of it being a specific git or svn branch.

Reply:
This was a hack introduced to help Spack sort versions since `develop` was treated as newer than all other versions. Now that #1983 is finally merged, this hack is no longer necessary, and both `master` and `develop` behave similarly.

Reply:
Huh, I didn't know about that.  I've been using develop in all the packages I work on for consistency in how to express a development version, regardless of the underlying source control mechanism or branch.

It seems though that develop still has special meaning in spack and always recognized as the most recent version.  So, given that, doesn't it still make the most sense to use develop as the version tag instead of the underlying git branch name? (I mean this mostly as a discussion on my viewpoint since I don't think spack has an established convention on that)

Reply:
> It seems though that develop still has special meaning in spack and always recognized as the most recent version.

Yes, and `master` also has special meaning in Spack, and is now recognized as the second most recent version. Let's put it this way. If a package has both a `develop` and `master` branch, and we want to add both versions, we can't name both `develop`. I think it makes the most sense to be clear about what branch you are getting through the version name, much like we are clear about release version numbers. The version is a _version_, not a sorting key.